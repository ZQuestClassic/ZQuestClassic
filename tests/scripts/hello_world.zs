global script Global
{
    void run()
    {
        int number = 123;
        printf("hello world: %d\n", number);
    }
}
