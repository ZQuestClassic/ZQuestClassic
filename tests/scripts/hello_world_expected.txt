Compiling 'hello_world.zs'
Pass 1: Parsing
Pass 2: Preprocessing
Pass 3: Registration
Pass 4: Analyzing Code
Pass 5: Checking code paths
Pass 6: Generating object code
Pass 7: Assembling
Success!
Compile finished with exit code '0' (success)

PUSHARGSV 0,0.0003; void run() Body Start
SETR d4,SP
SETV d2,123
STORED d2,1
ALLOCATEMEMV d2,17
STORED d2,0
WRITEPODSTRING d2,"hello world: %d\n"
LOADD d2,0; InlineFunc[void printf(char32)] Params Start
PUSHR d2
LOADD d2,1
PUSHVARGR d2; InlineFunc[void printf(char32)] Params End
PRINTFVARG; InlineFunc[void printf(char32)] Body Start
POP d5; InlineFunc[void printf(char32)] Body End
LOADD d3,0
DEALLOCATEMEMR d3
POPARGS d5,0.0003
QUIT; void run() Body End


QUIT; void run() Body